### PR TITLE
consensus/pbft: add unit tests for handle prepare

### PR DESCRIPTION
**Pull request by bailantaotao**

### CODE REVIEW

File: consensus/pbft/core/prepare_test.go [hunk 1]
```diff
@@ -18,19 +18,188 @@ package core
...
+				sys := NewTestSystemWithBackend(N, F)
+
+				for i, backend := range sys.backends {
+					c := backend.engine.(*core)
+
+					if i == 0 {
+						// replica 0 is primary
+						c.subject = expectedSubject
+						c.state = StatePreprepared
+						continue
```
Comment (yutelin) [inline consensus/pbft/core/prepare_test.go]: Just curious. Why do you prefer `continue` to `if else`?
`continue` is less readable to me.

Comment (bailantaotao) [inline consensus/pbft/core/prepare_test.go]: 因為看到`continue`之後我就可以知道不會有邏輯需要再被處理，不用再往下看

不過經你一說我覺得我的寫法沒有達到上述要求XD 我會再修一下

Comment (bailantaotao) [inline consensus/pbft/core/prepare_test.go]: fixed in 3cc332d

File: consensus/pbft/core/prepare_test.go [hunk 2]
```diff
@@ -18,12 +18,180 @@ package core
...
 import (
 	"math/big"
+	"reflect"
 	"testing"
 
 	"github.com/ethereum/go-ethereum/consensus/pbft"
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
+func TestCore_handlePrepare(t *testing.T) {
```
Comment (bailantaotao) [inline consensus/pbft/core/prepare_test.go]: fixed in 9d2ee18

**Pull request merged**